src/yunohost/app.py

```
@@ -829,12 +829,110 @@ def app_makedefault(auth, app, domain=None):
         raise MoulinetteError(errno.EPERM,
                               m18n.n('ssowat_persistent_conf_write_error', error=e.strerror))
 
+    os.system('chmod 644 /etc/ssowat/conf.json.persistent')
+
+    logger.success(m18n.n('ssowat_conf_updated'))
+
+
+def app_removedefault(auth, app, domain=None):
+    """
+    Remove the default redirection to an app on a domain
+
+    Keyword argument:
+        app
+        domain
+
+    """
+    from yunohost.domain import domain_list
+
+    app_settings = _get_app_settings(app)
+    app_domain = app_settings['domain']
+    app_path = app_settings['path']
+
+    if domain is None:
+        domain = app_domain
+    elif domain not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL, m18n.n('domain_unknown'))
+
+    if '/' in app_map(raw=True)[domain]:
+        raise MoulinetteError(errno.EEXIST,
+                              m18n.n('app_location_already_used'))
+
+    try:
+        with open('/etc/ssowat/conf.json.persistent') as json_conf:
+            ssowat_conf = json.loads(str(json_conf.read()))
+    except ValueError as e:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('ssowat_persistent_conf_read_error', error=e.strerror))
+    except IOError:
```

We should display a warning, never silencly hide errors.